packages/grid/stories/grid-cellValidation.stories.tsx

```diff
@@ -0,0 +1,335 @@
+);
+const formatter = new Intl.NumberFormat("en-US", {
+  style: "currency",
+  currency: "USD",
+});
+const getTotal = (r: RowExample) =>
+  formatter.format(r.amount * r.price * discountMap.get(r.discount)!);
+
+const useExampleDataSource = () => {
+  const [rows, setRows] = useState<RowExample[]>([
+    {
+      id: "Row1",
+      name: faker.commerce.productName(),
+      description: faker.commerce.productDescription(),
+      amount: faker.datatype.number({ min: 1, max: 100 }),
+      price: faker.datatype.number({ min: 10, max: 2000 }),
+      discount: "-",
+      status: "error",
+    },
+    {
+      id: "Row2",
+      name: faker.commerce.productName(),
+      description: faker.commerce.productDescription(),
+      amount: faker.datatype.number({ min: 1, max: 100 }),
+      price: faker.datatype.number({ min: 10, max: 2000 }),
+      discount: "-",
+    },
+    {
+      id: "Row3",
+      name: faker.commerce.productName(),
+      description: faker.commerce.productDescription(),
+      amount: faker.datatype.number({ min: 1, max: 100 }),
+      price: faker.datatype.number({ min: 10, max: 2000 }),
+      discount: "-",
+      status: "warning",
+    },
+    {
+      id: "Row4",
+      name: faker.commerce.productName(),
+      description: faker.commerce.productDescription(),
+      amount: faker.datatype.number({ min: 1, max: 100 }),
+      price: faker.datatype.number({ min: 10, max: 2000 }),
+      discount: "-",
+    },
+    {
+      id: "Row5",
+      name: faker.commerce.productName(),
+      description: faker.commerce.productDescription(),
+      amount: faker.datatype.number({ min: 1, max: 100 }),
+      price: faker.datatype.number({ min: 10, max: 2000 }),
+      discount: "-",
+      status: "success",
+    },
+  ]);
+
+  const setName = useCallback(
+    (row: RowExample, rowIndex: number, value: string) => {
+      setRows((x) => {
+        x = [...x];
+        x[rowIndex] = { ...x[rowIndex], name: value };
+        return x;
+      });
+    },
+    [setRows]
+  );
+
+  const setPrice = useCallback(
+    (row: RowExample, rowIndex: number, value: string) => {
+      setRows((x) => {
+        x = [...x];
+        x[rowIndex] = { ...x[rowIndex], price: Number.parseFloat(value) };
+        return x;
+      });
+    },
+    [setRows]
+  );
+
+  const setAmount = useCallback(
+    (row: RowExample, rowIndex: number, value: string) => {
+      setRows((x) => {
+        x = [...x];
+        x[rowIndex] = { ...x[rowIndex], amount: Number.parseInt(value) };
+        return x;
+      });
+    },
+    [setRows]
+  );
+
+  const setDiscount = useCallback(
+    (row: RowExample, rowIndex: number, value: string) => {
+      setRows((x) => {
+        x = [...x];
+        x[rowIndex] = { ...x[rowIndex], discount: value };
+        return x;
+      });
+    },
+    [setRows]
+  );
+
+  return { rows, setAmount, setName, setDiscount, setPrice };
+};
+
+export const CellValidation: Story = () => {
```

Comment:
Same as the column sorting, could you add an example showing async/server validation?

Reply:
@joshwooding ok this has now been updated with an example of async validation using yup which would be the common scenario for real time editing. I could also add a mock server validation example but I think it would be almost an exact copy paste but instead of using yup I would be using a fake fetch. So I don't see a point.